app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsources/derivepublickey/DerivePublicKeyViewModel.kt

```diff
@@ -46,19 +44,15 @@ class DerivePublicKeyViewModel @Inject constructor(
         derivePublicKeyJob = viewModelScope.launch {
             input = accessFactorSourcesUiProxy.getInput() as AccessFactorSourcesInput.ToDerivePublicKey
             when (input.factorSource) {
-                is LedgerHardwareWalletFactorSource -> {
-                    if (ensureBabylonFactorSourceExistUseCase.babylonFactorSourceExist()) {
-                        derivePublicKey().onSuccess {
-                            sendEvent(Event.AccessingFactorSourceCompleted)
-                        }
-                    } else {
-                        // 1st account created with ledger, so we need to create BDFS too and authenticate first
-                        sendEvent(Event.RequestBiometricPrompt)
+                is FactorSource.Ledger -> {
+                    derivePublicKey().onSuccess {
+                        sendEvent(Event.AccessingFactorSourceCompleted)
                     }
                 }
 
-                is DeviceFactorSource,
-                null -> {
+                is FactorSource.Device -> if (input.isBiometricsProvided) {
```

Comment:
not sure if this is fully safe. What if keystore access window of 30s will be exceeded? In which scenario this flow path is executed?

Reply:
This was the smallest change I could add to this flow to bridge the difference between android and sargon. To explain a bit how it works:

**New profile**
1. User is choosing to create an account when no wallet has been created yet.
2. The user types the name of the account and chooses whether to use Ledger or not.
3. **Upon Continue click** we request for biometrics in order to create the BDFS and then ask to derive the public key.
4. In this scenario since biometrics is already given, immediately the mnemonic is read again in order to derive the public key for the new account. So we should not ask the user for biometrics again. 

**Existing Profile**
1. The user does the same process and clicks Continue
2. Then we just ask to derive public keys with no biometrics thus the `isBiometricsProvided` is `false`.
3. The `DerivePublicKeyViewModel` itself asks for biometrics in this case.

Reply:
I had the same question but this 
> Upon Continue click we request for biometrics in order to create the BDFS and then ask to derive the public key.

answers it and we are safe to go